tests/internal/sriovoperator/sriovenv.go

```diff
@@ -42,3 +47,144 @@ func IsSriovDeployed(apiClient *clients.Settings, sriovOperatorNamespace string)
 
 	return nil
 }
+
+// WaitForSriovStable waits until all the SR-IOV node states are in sync.
+func WaitForSriovStable(apiClient *clients.Settings, waitingTime time.Duration, sriovOperatorNamespace string) error {
+	klog.V(90).Infof("Waiting for SR-IOV become stable.")
+
+	networkNodeStateList, err := sriov.ListNetworkNodeState(apiClient, sriovOperatorNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to fetch nodes state %w", err)
+	}
+
+	if len(networkNodeStateList) == 0 {
+		return nil
+	}
+
+	for _, nodeState := range networkNodeStateList {
+		err = nodeState.WaitUntilSyncStatus("Succeeded", waitingTime)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// WaitForSriovAndMCPStable waits until SR-IOV and MCP stable.
+func WaitForSriovAndMCPStable(
+	apiClient *clients.Settings,
+	waitingTime,
+	stableDuration time.Duration,
+	mcpName,
+	sriovOperatorNamespace string) error {
+	klog.V(90).Infof("Waiting for SR-IOV and MCP become stable.")
+	time.Sleep(10 * time.Second)
+
+	err := WaitForSriovStable(apiClient, waitingTime, sriovOperatorNamespace)
+	if err != nil {
+		return err
+	}
+
+	err = cluster.WaitForMcpStable(apiClient, waitingTime, stableDuration, mcpName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateSriovPolicyAndWaitUntilItsApplied creates SriovNetworkNodePolicy and waits until
+// it's successfully applied.
+func CreateSriovPolicyAndWaitUntilItsApplied(
+	apiClient *clients.Settings,
+	workerLabel,
+	sriovOperatorNamespace string,
+	sriovPolicy *sriov.PolicyBuilder,
+	timeout,
+	stableDuration time.Duration) error {
+	klog.V(90).Infof("Creating SriovNetworkNodePolicy %s and waiting until it's successfully applied.",
+		sriovPolicy.Definition.Name)
+
+	_, err := sriovPolicy.Create()
+	if err != nil {
+		return err
+	}
+
+	err = WaitForSriovAndMCPStable(
+		apiClient, timeout, stableDuration, workerLabel, sriovOperatorNamespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RemoveSriovConfigurationAndWaitForSriovAndMCPStable removes all SR-IOV networks
+// and policies in SR-IOV operator namespace.
+func RemoveSriovConfigurationAndWaitForSriovAndMCPStable(
+	apiClient *clients.Settings,
+	workerLabel,
+	sriovOperatorNamespace string,
+	mcoTimeout,
+	timeout time.Duration) error {
+	klog.V(90).Infof("Removing all SR-IOV networks and policies")
+
+	err := RemoveAllSriovNetworks(apiClient, sriovOperatorNamespace, timeout)
```

Comment:
Can you append this error and wait for RemoveAllPoliciesAndWaitForSriovAndMCPStable to run? . Otherwises there are policies that will not be removed causing issues in tests running afterwards